code-challenges.test.js

```diff
@@ -15,35 +15,136 @@
+
+const colors1 = ["purple", "blue", "green", "yellow", "pink"];
+    newArr = colors1.shift()
+    
+    for (let i = colors1.length -1; i > 0; i--) {
+      let j = Math.floor(Math.random() * i)
+      let k = colors1[i]
+      colors1[i] = colors1[j]
+      colors1[j] = k
+    }
+    console.log(colors1)
+
+const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"];
+    newArr = colors2.shift()
+
+for (let i = colors2.length -1; i > 0; i--) {
+  let j = Math.floor(Math.random() * i)
+  let k = colors2[i]
+  colors2[i] = colors2[j]
+  colors2[j] = k
+}
+console.log(colors2)
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const nums1 = [3, 56, 90, -8, 0, 23, 6]
+//const nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
-const nums2 = [109, 5, 9, 67, 8, 24]
-// Expected output: [5, 109]
-
+//const nums2 = [109, 5, 9, 67, 8, 24]
+//Expected output: [5, 109]
+describe('leastToGreatest', () => {
+    const expectedNums = [-8, 90];
+    it('takes an array of numbers and returns an array of the minimum and maximum numbers in that order', () => {
+        expect([3, 56, 90, -8, 0, 23, 6]).toEqual(expect.leastToGreatest(expectedNums))
+    });
+    it('does not match if received has other values besides [-8, 90]', () => {
+        expect([3, 56, 0, 23, 6]).not.toEqual(expect.leastToGreatest(expectedNums))
+    });
+});
+
+// TypeError: expect.leastToGreatest is not a function
 
 // b) Create the function that makes the test pass.
 
+//PseudoCode:
+//using the .sort() method to put the array from least to greatest value then .filter() to return those values in a seperate array. Or after sorting the values from least to greatest, have the function return the zero index and the last value possibly using .pop() and .shift()???
+
+//Parameters: Input:const nums1 = [3, 56, 90, -8, 0, 23, 6], Expected output: [-8, 90]
+//const nums2 = [109, 5, 9, 67, 8, 24], Expected output: [5, 109]
+
+// function leastToGreatest(array) {
+//             return nums1.sort(function(a, b){return a-b}).pop()
+//         }  
+//         console.log(leastToGreatest(nums1)) //90
+//     function leastToGreatest(array) {
+//             return nums1.sort(function(a, b){return a-b}).shift()
+//         }
+            
+//         console.log(leastToGreatest(nums1)) // -8
+//I could not figure out how to get these two results in the same function. I did try to make different variables and join the results that way but did not get very far. I was able to get the .sort() method down but it's just calling on the first and last values. I tried manipulating the return portion but was not sure if I had to include a for loop to access those indexes. Brining all the pieces together and writing it correctly is still a struggle for me. 
```

Comment:
If you could leave all of your approaches in here, I would appreciate it.  I do like your thought process for this problem.  You could assign the sorted array to a variable, and as you mentioned before making two different variables but instead of joining them, simply placing them into a new array.  Ex: [least, greatest] where least was assigned to the first value in the sorted array and greatest is the last value in the sorted array.